app/controllers/home_controller.rb

```diff
@@ -1,16 +1,12 @@
 class HomeController < ApplicationController
   def home
-    @users = User.all
-    @projects = Project.all
-    @samples = Sample.all
     @output_data = {}
+    @samples = Sample.paginate(page: params[:page]).order('created_at DESC')
 
-    @pipeline_outputs = PipelineOutput.order("created_at DESC")
-
-    @pipeline_outputs.each do |output|
-      sample_info =  output.sample
-      project_info = output.sample.project
-      @output_data['sample_info'] = sample_info
+    @samples.each do |output|
+      pipeline_output_info = output.pipeline_outputs.order('created_at').last
```

Comment:
this will break if there's no pipeline_output for a sample yet.  I think you will still show something on the page just say results are pending as shown currently.